src/main/scala/pureconfig/ConfigFieldMapping.scala

```diff
@@ -0,0 +1,24 @@
+package pureconfig
+
+trait ConfigFieldMapping[T] {
```

Comment:
I was thinking of an alternative to allow customizing the field mappings without having to provide implicit options while also keeping the flexibility to have different mapping rules for different types, and I think there's a better solution for it than providing a `String => String` function in the `loadConfig` method.

I think we can provide utilities to create `ConfigConvert` instances using a `ConfigFieldMapping` (which will probably become untyped then). That way, we can provide a system-wide conversion with a generic `ConfigConvert` and still customize more specific types.

@melrief: What do you think? If you like that idea, I can adapt this pull request.